src/test/java/de/wirvsvirus/testresult/backend/service/TestResultPushServiceTest.java

```diff
@@ -0,0 +1,105 @@
+package de.wirvsvirus.testresult.backend.service;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import de.wirvsvirus.testresult.backend.model.TestResult;
+import de.wirvsvirus.testresult.backend.model.TestResult.Result;
+
+@ExtendWith(MockitoExtension.class)
+public class TestResultPushServiceTest {
+
+	@Mock
+	private SmsServiceProvider smsService;
+
+	@Mock
+	private EmailService emailService;
+
+	@InjectMocks
+	TestResultPushService service;
+
+	@Test
+	public void sendEmailForNegativeResult() {
+
+		TestResult tr = new TestResult();
+		tr.setContact("mail@test.com");
+		tr.setStatus(Result.NEGATIVE);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertTrue(sentFlag),
+				() -> verify(emailService, times(1)).sendMail(any()),
+				() -> verify(smsService, never()).sendNegativeResultSms(any()));
+	}
+
+	@ParameterizedTest
+	@EnumSource(Result.class)
+	public void doNotSendEmailSentForNonNegativeResult(Result nonNegative) {
+
+		if (nonNegative.equals(Result.NEGATIVE))
+			return;
+
+		TestResult tr = new TestResult();
+		tr.setContact("mail@test.com");
+		tr.setStatus(nonNegative);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertFalse(sentFlag),

```

Comment:
I think message should be `"push isn't done and flag indicates it"`

Reply:
done